cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -102,7 +106,12 @@ public InMemoryConfigurator(CConfiguration cConf, Id.Namespace appNamespace, Id.
     SettableFuture<ConfigResponse> result = SettableFuture.create();
 
     try {
-      Object appMain = artifactClassLoader.loadClass(appClassName).newInstance();
+      Object appMain = impersonator.doAs(artifactId.getNamespace().toEntityId(), new Callable<Object>() {
```

Comment:
done; weird that we maintain namespace as well as artifactId.
@albertshau any idea why that's necessary?